### DISCUSSION TITLE
Can I flash a Realtek RTL8168 (rev 15) with iPXE?

### DISCUSSION THREAD

**Discussion by Saul5662** [ipxe/ipxe#1086] [Q&A]
I have tried running flashrom with `nicrealtek` but I keep getting no pci devices. Is it possible to flash it?

**Comment by NiKiZe** [comment]
Rtl8168 is most of the time an onboard device and have no flash connected. However, if your chip does have flash connected you might be.

**Reply by Saul5662** [reply]
Ah okay. Is it possible to modify a BIOS image to enable iPXE? I have an MSI B450M VDH-PRO MAX and the BIOS images are in a .BL0 format?

EDIT: Changed BK0 to BL0 (wrong file name xD)

**Reply by NiKiZe** [reply]
Yes you can modify BIOS flash.
Is it safe? You could brick your mobo.
Is it easy to do? Sometimes, but it all depends on which tools are available.

Some tools are leaked, and not really safe. You will probably have to look into "bios modification forums" to get further on that path.

**Reply by Saul5662** [reply]
Ah okay. Im trying to go down the option rom path. `.efirom` files from iPXE builds are option rom compatible right?

**Reply by NiKiZe** [reply]
It is for NIC ROM yes, available options somewhat documented at https://ipxe.org/appnote/buildtargets#boot_type

**Reply by Saul5662** [reply]
Hm okay, Can I insert it into my BIOS replacing the PXE DXE Driver/adding it as a new DXE Driver?
